src/lightning/fabric/fabric.py

```diff
@@ -601,10 +601,19 @@ def init(self) -> Generator:
 
     @contextmanager
     def init_module(self) -> Generator:
-        """Convenience context manager that will call :meth:`efficient_init` and :meth:`sharded_model`.
+        """Instantiate the model and its parameters under this context manager to reduce peak memory usage.
 
-        It is recommended to instantiate your modules under this.
+        The parameters get created on the device and with the right data type right away without wasting memory being
+        allocated unnecessarily. The automatic device placement under this context manager is only supported with
+        PyTorch 2.0 and newer.
         """
+        if not _TORCH_GREATER_EQUAL_2_0 and self.device.type != "cpu":
+            rank_zero_warn(
+                "`Fabric.init_module()` or `Fabric.init()` can't place the model parameters on the device directly"
```

Comment:
This references `fabric.init()` too.

If you want to put this here, then it would need to be added to `fabric.init()` too and to `strategy.init_context()` with the customized error message for each

Reply:
That's right. I updated the message.